framework/include/loops/ThreadedElementLoopBase.h

```diff
@@ -172,66 +174,73 @@ ThreadedElementLoopBase<RangeType>::operator()(const RangeType & range, bool byp
 
-      const Elem * elem = *el;
+      pre();
 
-      preElement(elem);
+      _subdomain = Moose::INVALID_BLOCK_ID;
+      _neighbor_subdomain = Moose::INVALID_BLOCK_ID;
+      typename RangeType::const_iterator el = range.begin();
+      for (el = range.begin(); el != range.end(); ++el)
+      {
+        if (!keepGoing())
+          break;
 
-      _old_subdomain = _subdomain;
-      _subdomain = elem->subdomain_id();
-      if (_subdomain != _old_subdomain)
-        subdomainChanged();
+        const Elem * elem = *el;
 
-      onElement(elem);
+        preElement(elem);
 
-      for (unsigned int side = 0; side < elem->n_sides(); side++)
-      {
-        std::vector<BoundaryID> boundary_ids = _mesh.getBoundaryIDs(elem, side);
+        _old_subdomain = _subdomain;
+        _subdomain = elem->subdomain_id();
+        if (_subdomain != _old_subdomain)
+          subdomainChanged();
 
-        if (boundary_ids.size() > 0)
-          for (std::vector<BoundaryID>::iterator it = boundary_ids.begin();
-               it != boundary_ids.end();
-               ++it)
-            onBoundary(elem, side, *it);
+        onElement(elem);
 
-        const Elem * neighbor = elem->neighbor_ptr(side);
-        if (neighbor != nullptr)
+        for (unsigned int side = 0; side < elem->n_sides(); side++)
         {
-          preInternalSide(elem, side);
-
-          _old_neighbor_subdomain = _neighbor_subdomain;
-          _neighbor_subdomain = neighbor->subdomain_id();
-          if (_neighbor_subdomain != _old_neighbor_subdomain)
-            neighborSubdomainChanged();
-
-          onInternalSide(elem, side);
+          std::vector<BoundaryID> boundary_ids = _mesh.getBoundaryIDs(elem, side);
 
           if (boundary_ids.size() > 0)
             for (std::vector<BoundaryID>::iterator it = boundary_ids.begin();
                  it != boundary_ids.end();
                  ++it)
-              onInterface(elem, side, *it);
+              onBoundary(elem, side, *it);
 
-          postInternalSide(elem, side);
-        }
-      } // sides
-      postElement(elem);
+          const Elem * neighbor = elem->neighbor_ptr(side);
+          if (neighbor != nullptr)
+          {
+            preInternalSide(elem, side);
 
-    } // range
+            _old_neighbor_subdomain = _neighbor_subdomain;
+            _neighbor_subdomain = neighbor->subdomain_id();
+            if (_neighbor_subdomain != _old_neighbor_subdomain)
+              neighborSubdomainChanged();
 
-    post();
+            onInternalSide(elem, side);
+
+            if (boundary_ids.size() > 0)
+              for (std::vector<BoundaryID>::iterator it = boundary_ids.begin();
+                   it != boundary_ids.end();
+                   ++it)
+                onInterface(elem, side, *it);
+
+            postInternalSide(elem, side);
+          }
+        } // sides
+        postElement(elem);
+
+      } // range
+
+      post();
+    }
+    catch (libMesh::LogicError & e)
```

Comment:
Ok I guess - I kind of wish this would have been in a separate PR (so that we could talk about why you did this)... but it's fine.